### PR TITLE
feat(mobile): search enhancement

### PR DESCRIPTION
**Pull request by alextran1502**
This PR adds the advanced search filters feature to the mobile app.

The result is fetched in batches of 1000, the infinite scrolling is janky due to the way the asset grid is built, so when a new batch is fetched, it rerenders the whole page. I might open a subsequent PR to address this issue

https://github.com/immich-app/immich/assets/27055614/a9110774-0778-4821-af5d-847091f3ab78

### CODE REVIEW

File: mobile/lib/modules/search/ui/explore_grid.dart [hunk 1]
```diff
@@ -56,9 +56,7 @@ class ExploreGrid extends StatelessWidget {
                       personName: content.label,
                     ),
                   )
-                : context.pushRoute(
-                    SearchResultRoute(searchTerm: 'm:${content.label}'),
-                  );
```
Comment (shenlong-tanwen) [inline mobile/lib/modules/search/ui/explore_grid.dart:61]: This stops the handling where tapping a curated place will display all assets from its location from working.

Comment (alextran1502) [reply mobile/lib/modules/search/ui/explore_grid.dart:61]: Correct, Also plan to fix it in a subsequent pr as well

Comment (alextran1502) [reply mobile/lib/modules/search/ui/explore_grid.dart]: actually fixed it in this PR as well

File: mobile/lib/modules/search/models/search_filter.dart [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'country': country,
+      'state': state,
+      'city': city,
+    };
+  }
+
+  factory SearchLocationFilter.fromMap(Map<String, dynamic> map) {
```
Comment (fyfrey) [inline mobile/lib/modules/search/models/search_filter.dart:37]: Do we use/need to/fromMap and to/fromJson at all for the filter classes?
I think they are never used.

**Review by fyfrey — APPROVED** [review]

File: mobile/lib/modules/search/models/search_filter.dart (hunk 2 above)
Comment (alextran1502) [reply mobile/lib/modules/search/models/search_filter.dart:37]: No, they are not used. It is auto-generated. I use this extension on vscode to help generate these classes

https://marketplace.visualstudio.com/items?itemName=dotup.dart-data-class-generator

**Pull request merged**